editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/preferences/ProjectTemplatesPreferencePage.java

```diff
@@ -0,0 +1,108 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package org.fusesource.ide.projecttemplates.preferences;
+
+import org.eclipse.core.runtime.IAdaptable;
+import org.eclipse.jface.preference.BooleanFieldEditor;
+import org.eclipse.jface.preference.FieldEditorPreferencePage;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchPreferencePage;
+import org.eclipse.ui.IWorkbenchPropertyPage;
+import org.fusesource.ide.projecttemplates.internal.Messages;
+import org.fusesource.ide.projecttemplates.internal.ProjectTemplatesActivator;
+
+/**
+ * This class represents a preference page that is contributed to the
+ * Preferences dialog. By subclassing <samp>FieldEditorPreferencePage</samp>, we
+ * can use the field support built into JFace that allows us to create a page
+ * that is small and knows how to save, restore and apply itself.
+ * <p>
+ * This page is used to modify preferences only. They are stored in the
+ * preference store that belongs to the main plug-in class. That way,
+ * preferences can be accessed directly via the preference store.
+ */
+public class ProjectTemplatesPreferencePage extends FieldEditorPreferencePage
+		implements IWorkbenchPreferencePage, IWorkbenchPropertyPage {
+
+	/**
+	 * Boolean field for turning staging repositories on/off.
+	 */
+	private BooleanFieldEditor enableStagingRepositorieslEditor;
+
+	/**
+	 * List field for modifying the list of staging repositories and their
+	 * order.
+	 */
+	private StagingRepositoryListEditor stagingRepositoriesListEditor;
+
+	/**
+	 * Constructor
+	 */
+	public ProjectTemplatesPreferencePage() {
+		super(GRID);
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see
+	 * org.eclipse.jface.preference.FieldEditorPreferencePage#createFieldEditors()
+	 */
+	@Override
+	public void createFieldEditors() {
+
+		this.enableStagingRepositorieslEditor = new BooleanFieldEditor(
+				ProjectTemplatesPreferencesConstants.ENABLE_STAGING_REPOSITORIES,
+				Messages.enableStagingRepositories_field, getFieldEditorParent());
+
+		addField(this.enableStagingRepositorieslEditor);
+
+		this.stagingRepositoriesListEditor = new StagingRepositoryListEditor(
+				ProjectTemplatesPreferencesConstants.STAGING_REPOSITORIES, Messages.stagingRepositoriesList_field,
+				getFieldEditorParent());
+
+		addField(this.stagingRepositoriesListEditor);
+
+		// Sets up the context sensitive help for this page
+		// PlatformUI.getWorkbench().getHelpSystem().setHelp(getFieldEditorParent(), "org.fusesource.ide.camel.editor.editorConfig");
```

Comment:
Not useless. We will need to hook help up to this preference page and this is a good indicator that remains to be done.